### ISSUE TITLE
Disable single price add workflow ?

### ISSUE THREAD

**Issue by raphodn** [openfoodfacts/open-prices-frontend#636]
### Story

Historically we started the project with a single price only workflow, where the user would scan the product, and then add its price & proof.

But we moved on to a proof-oriented workflow, add another workflow where the user first takes the proof photo (a receipt, or in a shop), and then starts adding product prices.

To simplify the UI, and match with the mobile, we should probably sunset the single price workflow.

### Why ?

- to simplify the code (no need to maintain both `AddPriceSingle` & `AddPriceMultiple`)
- to simplify the UI : 1 single "multiple price" workflow with proofs first

### Impact

The "single" price workflow is currently used when the user wants to add a price directly from a product or proof page

### Todo

- [x] hide the button on the "Price add" home page (users have only the choice to add multiple prices)
- [x] Proof detail: route the "Add a price" button to the multiple price workflow (and pre-select the proof)
- [ ] Product detail: route the "Add a price" button to the multiple price workflow (and pre-fill the product ? we still need to ask for a proof first)

**Comment by MrDemocracy** [comment]
Some feedback after using the new price add workflow for a bit:

- There is no done button when selecting a recent proof instead of uploading a new image, it's also not possible to use the previous page button in the browser to go back to the previous page anymore.
- The blue 'Add new price' button on the price add page seems unnecessary. I think it would be better if it was just replaced with the price upload form instead of having to click it for the first price entry each time.
- This could probably be a separate issue, but focusing text input fields when going to the next step of the form or e.g. visiting the search page would be nice for convenience. For the price add page it would be to focus the price field after clicking upload or or selecting the proof image.
- The `.v-overlay__scrim` element above the image makes it harder to see the price tags if you're looking at the image directly from the form to add prices.
- This was maybe always the case, but I cannot click the 'prices' count below the proof image to open the proof page from the price add page.

**Comment by odin-h** [comment]
Some feedback after using the new price add workflow for a bit:

- There is no done button when selecting a recent proof instead of uploading a new image, it's also not possible to use the previous page button in the browser to go back to the previous page anymore.
- The blue 'Add new price' button on the price add page seems unnecessary. I think it would be better if it was just replaced with the price upload form instead of having to click it for the first price entry each time.
- This could probably be a separate issue, but focusing text input fields when going to the next step of the form or e.g. visiting the search page would be nice for convenience. For the price add page it would be to focus the price field after clicking upload or or selecting the proof image.
- The `.v-overlay__scrim` element above the image makes it harder to see the price tags if you're looking at the image directly from the form to add prices.
- This was maybe always the case, but I cannot click the 'prices' count below the proof image to open the proof page from the price add page.

**Comment by raphodn** [comment]
thanks a lot for the feedback !!

> Seems the done button is only hidden until you have added a new price

you're right there should be a way to always "exit" the page, will put it back.

> I think it would be better if it was just replaced with the price upload form instead of having to click it for the first price entry each time

good idea. maybe we should also add a message explaining that adding prices will be possible only once the proof is selected/uploaded.

> but focusing text input fields when going to the next step

i usually scan the barcode first, so if we were to focus the price field the keyboard would popup..

> overlay doesn't allow to see the image

that's something we need to solve somehow, being able to zoom into the image to help for the price additions :+1: 

> clicking the proof price count from the price add page

yes it's on purpose, the proof is "readonly" to avoid changing the page. but i'm thinking of fetching and showing the existing prices, its on the todo

**Comment by odin-h** [comment]
>you're right there should be a way to always "exit" the page, will put it back.

Might be a good idea to put the button on the opposite side of the upload button so you don't accidentally click the done button instead of upload after selecting an image. Alternatively, it could prompt you with the browser native "leave site?" popup using the `beforeunload` even listener warning you about still having unsaved work.

>good idea. maybe we should also add a message explaining that adding prices will be possible only once the proof is selected/uploaded.

Yeah, maybe, but isn't this implied by the red `Add a proof` message? The price upload form should also not be inserted until the proof has been added, same way the blue button works currently.

>i usually scan the barcode first, so if we were to focus the price field the keyboard would popup..

Oh, right. Forgot about how that would affect mobile.

>that's something we need to solve somehow, being able to zoom into the image to help for the price additions 👍

My workaround has been to just AdBlock the overlay and use a browser extension that zooms the image and follow your cursor when you hover over the image, but I still find myself preferring to compare it against the original image from my photo album due to the compression making the price tags harder to read sometimes. However, that is probably more due to  an issue with my phone's camera quality than with the compression.

>yes it's on purpose, the proof is "readonly" to avoid changing the page. but i'm thinking of fetching and showing the existing prices, its on the todo :)

Okay, I see... Maybe in addition to fetching existing prices then, the `beforeunload` even listener could be used here too in order to prevent you from losing unsaved work? That way you could still click it like normally when you don't have any unsaved work.

One reason I use the recent proof feature sometimes is when I have added too many prices for the proof image to still show on the left side so I don't have to scroll up and down so much when adding more prices. However, this could probably be solved by sticking the image preview to the page, so the list of prices on the right side scrolls independently of the proof container? That probably wouldn't work on the mobile version though.

**Comment by raphodn** [comment]
Closing as this page has been moved to the Experiment tab instead
